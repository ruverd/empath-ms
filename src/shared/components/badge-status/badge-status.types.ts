import { type VariantProps } from "class-variance-authority";

import type { badgeStatusVariants } from "./badge-status";

export interface BadgeStatusProps
  extends VariantProps<typeof badgeStatusVariants> {}
