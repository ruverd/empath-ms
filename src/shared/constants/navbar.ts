export const NAV_ITEMS = [
  {
    label: "Dashboard",
    href: "/",
  },
  {
    label: "Members",
    href: "/members",
  },
  {
    label: "Courses",
    href: "/courses",
  },
  {
    label: "Learning Pathways",
    href: "/learning",
  },
  {
    label: "Live Streams",
    href: "/live",
  },
];
