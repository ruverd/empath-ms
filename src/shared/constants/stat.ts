export enum STAT_STATUS {
  ACTIVE = "ACTIVE",
  COMING_SOON = "COMING_SOON",
}
