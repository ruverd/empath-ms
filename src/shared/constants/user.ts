import type { User } from "../types/user.types";

export const USER_AUTHENTICATED = {
  name: "Jason Momoa",
  email: "jason.momoa@example.com",
  avatar: undefined,
} satisfies User;
