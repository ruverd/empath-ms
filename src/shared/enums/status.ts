export enum Status {
  ON_TRACK = "On Track",
}
