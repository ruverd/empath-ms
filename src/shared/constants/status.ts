import { Status } from "../enums/status";

export const STATUS = {
  [Status.ON_TRACK]: "On Track",
};
