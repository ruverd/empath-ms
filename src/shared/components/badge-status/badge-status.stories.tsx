import type { Meta, StoryObj } from "@storybook/react";

import { Status } from "@/shared/enums/status";

import { BadgeStatus } from "./badge-status";

const meta = {
  title: "Components/BadgeStatus",
  component: BadgeStatus,
  parameters: {
    layout: "centered",
  },
  decorators: [
    (Story) => (
      <div className="flex justify-center items-center gap-2 size-80 bg-primary-indigo">
        <Story />
      </div>
    ),
  ],
  tags: ["autodocs"],
} satisfies Meta<typeof BadgeStatus>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    variant: Status.ON_TRACK,
  },
};
