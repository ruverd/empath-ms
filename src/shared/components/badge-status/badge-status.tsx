import { cva } from "class-variance-authority";

import { STATUS } from "@/shared/constants/status";
import { Status } from "@/shared/enums/status";
import { cn } from "@/shared/lib/utils";

import type { BadgeStatusProps } from "./badge-status.types";

export const badgeStatusVariants = cva("size-2 rounded-full", {
  variants: {
    variant: {
      [Status.ON_TRACK]: "bg-secondary-green",
    },
  },
  defaultVariants: {
    variant: Status.ON_TRACK,
  },
});

export const BadgeStatus = ({ variant }: BadgeStatusProps) => {
  return (
    <div className="flex items-center gap-2">
      <div className={cn(badgeStatusVariants({ variant }))} />
      <span className="text-xs font-medium text-primary-cloud">
        {STATUS[variant!]}
      </span>
    </div>
  );
};
