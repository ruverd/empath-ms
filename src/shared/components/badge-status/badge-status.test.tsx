import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { STATUS } from "@/shared/constants/status";
import { Status } from "@/shared/enums/status";

import { BadgeStatus } from "./badge-status";

describe("BadgeStatus", () => {
  it("should renders with ON_TRACK variant", () => {
    render(<BadgeStatus variant={Status.ON_TRACK} />);

    const statusText = screen.getByText(STATUS[Status.ON_TRACK]);
    expect(statusText).toBeInTheDocument();
  });

  it("should renders the status indicator dot", () => {
    render(<BadgeStatus variant={Status.ON_TRACK} />);

    const container = screen.getByText(STATUS[Status.ON_TRACK]).parentElement;
    const dot = container?.firstChild;
    expect(dot).toHaveClass("size-2", "rounded-full", "bg-secondary-green");
  });

  it("should match snapshot", () => {
    const { container } = render(<BadgeStatus variant={Status.ON_TRACK} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
