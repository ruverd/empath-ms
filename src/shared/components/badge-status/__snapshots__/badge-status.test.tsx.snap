// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BadgeStatus > should match snapshot 1`] = `
<div
  class="flex items-center gap-2"
>
  <div
    class="size-2 rounded-full bg-secondary-green"
  />
  <span
    class="text-xs font-medium text-primary-cloud"
  >
    On Track
  </span>
</div>
`;
